### DISCUSSION TITLE
💻童年之window XP(淺談虛擬機與操作)(原創文章)

### DISCUSSION THREAD

**Discussion by JunTingLin** [1101-GS4719-C/Python-Community-GS4719-C#370] [General]
# :loudspeaker:``分享虛擬機操作``

## 前言
今天上課看到老師有用VM，可能老師的本機實體點腦有裝過python，故老師用虛擬機示範

這裡來跟各位分享VM的強大與操作🎉

## 甚麼叫VM~它很強嗎
VM虛擬機器是實體電腦的虛擬呈現或模擬環境
可以在虛擬機內安裝不同作業系統，以下圖為例因為之前上課需要安裝了Ubuntu作業系統、window server 2016，也裝過去年底剛出的win11在虛擬機內，
![image](https://user-images.githubusercontent.com/92431095/155330643-5b2adbea-9de3-420f-aebe-7f72a11d6ae7.png)

一台實體電腦可以安裝這麼多系統，四八四很酷呢?(❁´◡`❁)

## 甚麼時候我會需要用到VM
今天假如想做些危險性操作，比如說上~~暗網逛逛、破解軟體~~  (誤~😁)，不想把你實體系統搞到爆掉🧨，導致無法回天的境界，那虛擬軟體一定非常適合

## 我來玩玩、回味童年
這邊出於好奇來回味一下我的童年👶，來裝裝window XP ~~(想必那熟悉的開機音樂會人無法自拔)~~

## 虛擬機有哪些?哪個好?
Oracle VM VirtualBox、QEMU、SolarWinds Virtualization Manager都是可在網路上蒐到且免費的虛擬機

以win電腦來說其實都踏大同小異，操作邏輯也都差不多

但我身邊的 鵬友們~~都推VMware Workstation

### 步驟1
到VMware官網下載軟體，無腦一直按下一步即可
![image](https://user-images.githubusercontent.com/92431095/155330766-3939927d-7526-40fe-9768-9cfe0ad5726c.png)

VMware Workstation非商業用途安裝免費版即可

### 步驟2
點選創建一個新的VM

### 步驟3
選擇典型安裝即可
![image](https://user-images.githubusercontent.com/92431095/155330788-be8282be-c172-4e64-bafd-7e54e65d9188.png)

### 步驟4
找到win XP光碟ISO映像檔，可以去網路上找找，並按開啟
![image](https://user-images.githubusercontent.com/92431095/155330844-d8737fd9-b7b1-4e6c-96b1-da63a3fb9a95.png)
![image](https://user-images.githubusercontent.com/92431095/155330879-d69c6f41-4a3d-4343-8809-8f1a65dad390.png)

### 步驟5
它會請你幫這個虛擬機命名，並且選擇安裝位置
![image](https://user-images.githubusercontent.com/92431095/155330900-2361696f-6175-45d0-a49d-38f845d5c74d.png)

### 步驟6
會問你需要給win XP多少容量，這邊可以事後在調都沒差
![image](https://user-images.githubusercontent.com/92431095/155330923-9dd45f00-5744-4189-910a-9a1650d71352.png)

但....要注意下面兩個核取方塊的區別

1. 代表這個虛擬機只認你這台電腦
2. 可以把系統winXP(整個工作資料夾)複製到其他台電腦(前提:其他台電腦有VMware Workstation)

選擇2當然比較好，但相對的占用容量就會比較大

### 步驟7
接續一直按下一步就會進入winXP的安裝程序，這邊要輸入序號
![image](https://user-images.githubusercontent.com/92431095/155330943-6f8fbbbf-3225-4366-9b77-3060ee43bcd5.png)

呵呵~~ winXP都多久以前了，網上序號一堆

### 步驟8
再次聽到熟悉的開機聲音~~爽
![image](https://user-images.githubusercontent.com/92431095/155330962-ab13ba6b-8f8e-4eb2-ae31-fd41093280ec.png)

## 補充
假如是mac系統的推薦parallel desktop (只不過要付費😥)

可以讓win跟mac有更好的相容

用過就回不去啦~~!_!

強烈推薦mac用戶一定要裝，當在灌win進parallel虛擬機時你會非常有感

`第一次發文🤣，不確定這樣的模式對不對，有錯誤或要更正的歡迎討論`

**Comment by chia-kai-chang** [comment]
同學你好棒，你獲得全班第一個👍🎉

**Comment by handsomeniko** [comment]
VM真的強~

**Comment by chupochun** [comment]
謝謝介紹VM 我以前從來沒聽過這個軟件

**Comment by pekoranoinu** [comment]
感謝大大為我們介紹VM~

**Comment by Ben6uo** [comment]
可惜的是 不是所有的東西都可以在虛擬機運行

**Comment by lx3211** [comment]
我的mac絕對是個坑貨
之前試過VM一直出包

**Comment by boson13579** [comment]
跟各位說，那個虛擬機的大小阿....別給太小....，調起來好麻煩，要先在vmware manager 裡面調(喔,對了,虛擬機要關機才能調,儲存狀態不行)，之後還要進到虛擬機裡面調(看裡面的作業系統是什麼,調的方式不一樣),這邊有一個沒在裡面調的,想說明明給空間了,卻還是超小...

**Comment by Tzuhsuannnnn** [comment]
系上的課，老師們很常都會提到虛擬機，一開始完全不知道這個要幹嘛，經過了一學期的使用才逐漸比較熟悉，透過你的文章也更了解整體的運作了!

**Comment by RogelioKG** [comment]
洛克人X5（2000），我的童年回憶。以前在鄉下和阿公阿嬤住，因為他們很早就要下田，早上五點就要出發，我就會趁那個時間，打開老家那台又胖又重的CRT螢幕、XP系統的電腦（CRT螢幕的刷新率堪比電競螢幕呢［滑稽🥴］），開始玩洛克人X5，小時候（那個時候大概5歲吧）因為不懂所以都在亂按，按著按著竟然還按的有聲有色（？），只不過遇到很難的BOSS還是會卡很久，我依稀記得有一隻我卡了一個月（原來這就是我的魂系啟蒙，恍然大悟）。

大概是高三那個時候，不知怎的突然和朋友聊起童年回憶，才突然想到這款已經要生灰塵的遊戲。因為一開始是用 Windows XP 運行的，所以當時就想應該是開 Windows XP 的虛擬機，然後在上面運行抓來的遊戲，結果卡在動畫 run 不起來。我後來爬文才發現，原來洛克人X5根本沒有在PC上架（想當然小時候電腦裡的那個是盜版）。後來改用PS 的虛擬機 ePSXe 之後成功運行。三小時破關（經歷過魂系的摧殘後，身法成長不少🥲）

**Comment by zhiron544** [comment]
關於虛擬主機我也有研究，不過是安裝在Android上的，可以使用lLimbo x86套件再加上Windows的輕量化版本就可以使用了，但問題是速度太慢需要再繼續化。
